lib/src/config.cc

```diff
@@ -305,6 +311,171 @@ namespace hocon {
         return long_list;
     }
 
+    duration config::get_duration(string const& path) const {
+        auto v = get_value(path);
+        if (auto d = dynamic_pointer_cast<const config_double>(v)) {
+            return convert(d->double_value(), time_unit::MILLISECONDS);
+        } else if (auto l = dynamic_pointer_cast<const config_long>(v)) {
+            return convert(l->long_value(), time_unit::MILLISECONDS);
+        } else if (auto i = dynamic_pointer_cast<const config_int>(v)) {
+            return convert(i->long_value(), time_unit::MILLISECONDS);
+        } else if (auto str = dynamic_pointer_cast<const config_string>(v)) {
+            return parse_duration(str->transform_to_string(), str->origin(), path);
+        } else {
+            throw bad_value_exception(*v->origin(), path, _("Value at '{1}' was not a number or string.", path));
+        }
+    }
+
+    double config::get_duration_as_double(string const& path, time_unit unit) const {
+        auto timespan = get_duration(path);
+        switch (unit) {
+            case time_unit::NANOSECONDS:
+                return (timespan.first * 1000000000.0) + timespan.second;
```

Comment:
It would be nice to throw an exception on overflow/underflow. See http://stackoverflow.com/questions/15655070/how-to-detect-double-precision-floating-point-overflow-and-underflow for some ideas.